### PR TITLE
add new plugin for handyvertrag.de

### PR DESCRIPTION
**Pull request by sdx23**
Adds a new plugin to download invoices and connection-reports from handyvertrag.de

What linter and style-checker should be used with which settings?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +110,8 @@ Commands:
   dkb       dkb.de with chipTAN QR (postbox)
   elster    elster.de with path to .pfx certfile as username (postbox)
   ing       banking.ing.de with photoTAN (postbox)
+  handyvertrag
+            handyvertrag.de (invoices)
```
Comment (heeplr) [inline README.md]: here we should also mention "(invoices, call logs)". I will add this to the other plugins as well.

File: docdl/plugins/handyvertrag.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                        'category': 'connection_report',
+                        'subject': description,
+                    })
+
+
+@click.command()
+@click.pass_context
+# pylint: disable=C0103
+def handyvertrag(ctx):
+    """service.handyvertrag.de (invoices/postbox)"""
```
Comment (heeplr) [inline docdl/plugins/handyvertrag.py]: same as above: "(invoices, call logs)"

File: docdl/plugins/handyvertrag.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    By.LINK_TEXT, 'Einzelverbindungsnachweis')
+            if len(link) == 1:
+                url = link[0].get_attribute("href")
+                idn = url.split('/')[-1]
+                yield docdl.Document(
+                    url=url,
+                    attributes={
+                        'id': idn,
+                        'date': docdl.util.parse_date(date),
+                        'category': 'connection_report',
```
Comment (heeplr) [inline docdl/plugins/handyvertrag.py]: i think "call log" would be the correct translation, but i'm not sure. The category can be plugin specific but I'd suggest trying to keep it consistent with documentation.

**Comment by heeplr** [comment]
> Adds a new plugin to download invoices and connection-reports from handyvertrag.de

Awesome! Thank you very much.

> What linter and style-checker should be used with which settings?

I'd suggest flake8 without settings (and inline overrides containing a reason for the override). Maybe globally allowing >80 char lines. I'm not sure, yet. I'll setup CI - can still be modified later.

File: docdl/plugins/handyvertrag.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        # Login failed
+        return self.webdriver.find_elements(
+            By.XPATH, "//img[contains(@alt, 'LOGOUT')]"
+        )
+
+    def logout(self):
+        # logoutbutton = self.webdriver.find_element(
+        #       By.XPATH, '//span[@id, "logoutLink"]/a')
+        # logoutbutton.click()
+        self.webdirver.get(self.URL_LOGOUT)
```
Comment (heeplr) [inline docdl/plugins/handyvertrag.py]: thats a typo

**Comment by heeplr** [comment]
> Adds a new plugin to download invoices and connection-reports from handyvertrag.de

Awesome! Thank you very much.

> What linter and style-checker should be used with which settings?

I'd suggest flake8 without settings (and inline overrides containing a reason for the override). Maybe globally allowing >80 char lines. I'm not sure, yet. I'll setup CI - can still be modified later.

EDIT: maybe pylint would be better suited

EDIT2: I've setup pylint + flake8 CI for pushes and pull-requests. I'll happily merge this PR when the issues above are fixed.

File: README.md (hunk 1 above)
Comment (sdx23) [reply README.md]: I've found "itemised bill", "itemised call record" and "itemisation report" for "Einzelverbindungsnachweis". "call log" translates back to "Anrufliste". Certainly anything is better than "connection report" :)

File: docdl/plugins/handyvertrag.py (hunk 4 above)
Comment (sdx23) [reply docdl/plugins/handyvertrag.py]: whoops.

File: README.md (hunk 1 above)
Comment (heeplr) [reply README.md]: i think "call record" would be something, anyone would associate with "EVN". I'll change it after merging.

**Pull request merged**